vmrunner/vmrunner.py

```diff
@@ -180,6 +180,136 @@ def name(self):
     def image_name(self):
         abstract()
 
+    def start_process(self, cmdlist):
+
+        if cmdlist[0] == "sudo": # and have_sudo():
+            print color.WARNING("Running with sudo")
+            self._sudo = True
+
+        # Start a subprocess
+        self._proc = subprocess.Popen(cmdlist,
+                                      stdout = subprocess.PIPE,
+                                      stderr = subprocess.PIPE,
+                                      stdin = subprocess.PIPE)
+
+        return self._proc
+
+
+# Ukvm Hypervisor interface
+class ukvm(hypervisor):
+
+    def __init__(self, config):
+        # config is not yet used for ukvm
+        super(ukvm, self).__init__(config)
+        self._proc = None
+        self._stopped = False
+        self._sudo = False
+        self._image_name = self._config if "image" in self._config else self.name() + " vm"
+
+        # Pretty printing
+        self.info = Logger(color.INFO("<" + type(self).__name__ + ">"))
+
+    def name(self):
+        return "Ukvm"
+
+    def image_name(self):
+        return self._image_name
+
+    def drive_arg(self):
+        return ["--disk=" + INCLUDEOS_HOME + "dummy.disk"]
```

Comment:
Some services need a real disk image, like https://github.com/hioa-cs/IncludeOS/tree/master/test/fs/integration/fat32.

Reply:
I see, I guess we can use vm.json to pass the disk and net parameters when not present defaults to dummy.disk and tap100. 
Also, instead of setting tap interface and creating dummy.disk in boot.py, we can move that to vmrunner.py where it can be configured based on vm.json

A sample vm.json
```
{
  "net" : "tap101"
  "disk"   : "another.disk"
}
``` 
Thoughts?

Reply:
Was thinking what to do about the TAPs. Qemu runs get a script that's used to create the TAP. In the case of solo5/ukvm, it seems that vmrunner should be the place where the TAPs are created (and the disk).